core/trino-main/src/main/java/io/trino/execution/executor/TaskExecutor.java

```diff
@@ -90,6 +91,8 @@
     private final int minimumNumberOfDrivers;
     private final int guaranteedNumberOfDriversPerTask;
     private final int maximumNumberOfDriversPerTask;
+    private final DistributionStat allSplitsSizeDistribution = new DistributionStat();
+    private long lastTimeSplitSizeWasRecorded;
```

Comment:
I've added. Yes, it is synchronized in every updating path. 
The reading is not synchronized intentionally in `TaskExecutor.getAllSplitSizeDistribution`. I could make it synchronized easily but I decided to not do it because I saw that we do it in some places for stats (looks for the comment in `OutputBuffer` for example). 
We sacrifice inconsistency over performance here - any sample of JMX attributes could introduce additional contention here.

Reply:
As we discussed with @lukasz-stec offline we can have internally inconsistent state here (because we have no lock on the `TaskExecutor.getAllSplitSizeDistribution` method) but we can accept it. It is thread-safe because of `StatDistribution` is thread-safe.